components/payments/cmd/connectors/internal/connectors/generic/client/errors.go

```diff
@@ -12,6 +12,10 @@ var (
 )
 
 func wrapError(err error, resp *http.Response) error {
+	if resp == nil {
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("%w: %w", err, ErrStatusCodeServerError)
```

since retrying would not be useful in this case

Reply:
looks like it's being caught and converted to `task.ErrRetryable`

Reply:
I added the new error and set it as retryable error in the relevant parts